src/wallet/wallet.cpp

```diff
@@ -4342,7 +4339,7 @@ CWallet* CWallet::CreateWalletFromFile(const std::string& name, const fs::path&
     }
 
     // Register with the validation interface. It's ok to do this after rescan since we're still holding cs_main.
-    RegisterValidationInterface(temp_wallet.release());
+    RegisterValidationInterface(walletInstance.get());
```

Comment:
When does this get unregistered? Seems like the validation interface queue may hold a pointer after the wallet is released.

Reply:
When `UnregisterAllValidationInterfaces` is called. Obviously `unloadwallet` RPC or the `Unload wallet` action in the UI will have to unregister.